src/metatensor/models/cli/train_model.py

```diff
@@ -182,70 +177,67 @@ def _train_model_hydra(options: DictConfig) -> None:
     )
     train_targets = read_targets(train_options["targets"])
     train_dataset = Dataset(train_structures, train_targets)
+    train_size = 1.0
 
     logger.info("Setting up test set")
     test_options = options["test_set"]
-    if not isinstance(test_options, float):
+
+    if isinstance(test_options, float):
+        test_size = test_options
+        train_size -= test_size
+
+        if test_size < 0 or test_size >= 1:
+            raise ValueError("Test set split must be between 0 and 1.")
+
+        generator = torch.Generator()
+        if options["seed"] is not None:
+            generator.manual_seed(options["seed"])
+
+        train_dataset, test_dataset = _train_test_random_split(
+            train_dataset=train_dataset,
+            train_size=train_size,
+            test_size=test_size,
+            generator=generator,
+        )
+    else:
         test_options = expand_dataset_config(test_options)
         test_structures = read_structures(
             filename=test_options["structures"]["read_from"],
             fileformat=test_options["structures"]["file_format"],
         )
         test_targets = read_targets(test_options["targets"])
         test_dataset = Dataset(test_structures, test_targets)
-        test_fraction = 0.0
         check_units(actual_options=test_options, desired_options=train_options)
-    else:
-        if test_options < 0 or test_options >= 1:
-            raise ValueError("Test set split must be between 0 and 1.")
-        test_fraction = test_options
 
     logger.info("Setting up validation set")
     validation_options = options["validation_set"]
-    if not isinstance(validation_options, float):
+    if isinstance(validation_options, float):
+        validation_size = validation_options
+        train_size -= validation_size
+
+        if validation_size < 0 or validation_size >= 1:
+            raise ValueError("Validation set split must be between 0 and 1.")
+
+        generator = torch.Generator()
+        if options["seed"] is not None:
+            generator.manual_seed(options["seed"])
+
+        train_dataset, validation_dataset = _train_test_random_split(
+            train_dataset=train_dataset,
+            train_size=train_size,
+            test_size=validation_size,
+            generator=generator,
+        )
```

Comment:
Careful: I don't know if this is doing what we want. For example, if the validation/test are 0.1/0.1, you're going to take out the test split first, then train is 0.9. Then, you take out the validation fraction and new `train_size` is 0.8 and `validation_size` is 0.1. Then you split your remaining training dataset according to `0.8` and `0.1` and you throw away some of it (0.1). Am I right?

Reply:
Good question, but I think we are doing this correctly. Inside the `_train_test_random_split ` we normalize the sizes to `1`. If we won't do this you were right but look at the example below. 

```python
from torch.utils.data import random_split
import torch


train, test = random_split(range(100), [0.9,0.1])

length = torch.tensor([0.8,0.1])
length /= sum(length)

train, validation = random_split(train, length)

print("train:",len(train.indices))
print("test:", len(test.indices))
print("validation:", len(validation.indices))
```

I am doing the normalization inside the `_train_test_random_split` function to save some code. But, if this is unclear we can either (a) document this or (b) move the part outside of the code.